src/server/routes/users.js

```diff
@@ -135,47 +138,98 @@ router.post('/create', adminAuthMiddleware('create a user.'), async (req, res) =
  * Route for updating user role
  */
 router.post('/edit', adminAuthMiddleware('update a user role'), async (req, res) => {
+	
 	const validParams = {
 		type: 'object',
-		required: ['users'],
+		required: ['user'],
 		properties: {
-			users: {
-				type: 'array',
-				items: {
-					type: 'object',
-					required: ['email', 'role'],
-					properties: {
-						email: {
-							type: 'string'
-						},
-						role: {
-							type: 'string',
-							enum: Object.values(User.role)
-						}
-
+			user: {
+				type: 'object',
+				required: ['id', 'email', 'role'],
+				properties: {
+					id: {
+						type: 'integer'
+					},
+					email: {
+						type: 'string'
+					},
+					role: {
+						type: 'string',
+						enum: Object.values(User.role)
+					},
+					password: {
+						type: 'string'
+					},
+					note: {
+						type: 'string'
 					}
 				}
 			}
 		}
 	};
+
 	if (!validate(req.body, validParams).valid) {
 		res.status(400).json({ message: 'Invalid params' });
 	} else {
+
 		try {
 			const conn = getConnection();
-			const { users } = req.body;
-			const minimumUser = users.find(user => user.role === User.role.ADMIN);
-			// This protects the database so that there will always be at least one admin during role updates.
-			if (minimumUser === undefined) {
-				res.sendStatus(400);
-			} else {
-				const roleUpdates = users.map(async user => await User.updateUserRole(user.email, user.role, conn));
-				await Promise.all(roleUpdates);
-				res.sendStatus(200);
+			const { user } = req.body;
+			const userBeforeChanges = await User.getByID(user.id,conn);
+			const numberOfAdmins = (await User.getNumberOfAdmins(conn)).count;
+			
+			// This protects the database so that there will always be at least one admin
+			if (numberOfAdmins < 2 && userBeforeChanges.role === 'admin' && user.role !== 'admin') {
```

Comment:
Not a big deal but the database calls and check for # admims could be moved inside this if so it only happens if a role change is being made.

Reply:
Got it. Done